### ISSUE TITLE
Fit time timer utility

### ISSUE THREAD

**Issue by wagner2010** [DistrictDataLabs/yellowbrick#439]
Add timer utility functionality to the library to adapt any learned `fit_time_` parameters:

```python
import time

from functools import wraps
from dateutil.relativedelta import relativedelta


def humanizedelta(*args, **kwargs):
    """
    Wrapper around dateutil.relativedelta (same construtor args) and returns
    a humanized string representing the delta in a meaningful way.
    """
    delta = relativedelta(*args, **kwargs)
    attrs = ('years', 'months', 'days', 'hours', 'minutes', 'seconds')
    parts = [
        '%d %s' % (getattr(delta, attr), getattr(delta, attr) > 1 and attr or attr[:-1])
        for attr in attrs if getattr(delta, attr)
    ]

    return " ".join(parts)


class Timer(object):
    """
    A context object timer. Usage:
        >>> with Timer() as timer:
        ...     do_something()
        >>> print timer.interval
    """

    def __init__(self, wall_clock=True):
        """
        If wall_clock is True then use time.time() to get the number of
        actually elapsed seconds. If wall_clock is False, use time.clock to
        get the process time instead.
        """
        self.wall_clock = wall_clock
        self.time = time.time if wall_clock else time.clock

    def __enter__(self):
        self.start    = self.time()
        return self

    def __exit__(self, type, value, tb):
        self.finish   = self.time()
        self.interval = self.finish - self.start

    def __str__(self):
        return humanizedelta(seconds=self.interval)


def timeit(func, wall_clock=True):
    """
    Times a function; returns a Timer along with the result.
    """
    @wraps(func)
    def timer_wrapper(*args, **kwargs):
        """
        Inner function that uses the Timer context object
        """
        with Timer(wall_clock) as timer:
            result = func(*args, **kwargs)

        return result, timer
    return timer_wrapper
```

**Comment by ZJPoh** [comment]
If no one else is working on this, I'd be happy to start working on this. This is my first time contributing so I might have some questions.

**Comment by rebeccabilbro** [comment]
Hey there @ZJPoh, that would be great! Let us @DistrictDataLabs/team-oz-maintainers  know if there's anything we can help with!

**Comment by bbengfort** [comment]
Closed by #493
